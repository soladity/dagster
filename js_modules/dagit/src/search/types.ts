export enum SearchResultType {
  Asset,
  Page,
  Pipeline,
  Repository,
  Run,
  Schedule,
  Sensor,
  Solid,
}

export type SearchResult = {
  key: string;
  label: string;
  description: string;
  href: string;
  type: SearchResultType;
  tags?: string;
};
