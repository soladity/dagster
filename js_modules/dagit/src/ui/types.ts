export type Spacing = 0 | 1 | 2 | 4 | 8 | 12 | 16 | 20 | 24 | 32 | 48 | 64;
export type DirectionalSpacing =
  | Spacing
  | {
      top?: Spacing;
      right?: Spacing;
      bottom?: Spacing;
      left?: Spacing;
      vertical?: Spacing;
      horizontal?: Spacing;
    };

export type AlignItems =
  | 'stretch'
  | 'center'
  | 'start'
  | 'end'
  | 'flex-start'
  | 'flex-end'
  | 'baseline';

export type FlexDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';
export type FlexProperties = {
  alignItems?: AlignItems;
  basis?: string;
  direction?: FlexDirection;
  display?: 'flex' | 'inline-flex';
  grow?: number;
  justifyContent?:
    | 'center'
    | 'start'
    | 'end'
    | 'flex-start'
    | 'flex-end'
    | 'left'
    | 'right'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | 'stretch';
  shrink?: number;
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
};

export type BorderSide = 'top' | 'right' | 'bottom' | 'left' | 'horizontal' | 'vertical' | 'all';
export type BorderWidth = 1 | 2;
export type BorderColor = 'light' | 'medium' | 'dark';
export type BorderSetting = {width: BorderWidth; color: BorderColor; side: BorderSide};
