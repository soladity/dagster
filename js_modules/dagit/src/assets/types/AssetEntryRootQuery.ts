// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssetKeyInput } from "./../../types/globalTypes";

// ====================================================
// GraphQL query operation: AssetEntryRootQuery
// ====================================================

export interface AssetEntryRootQuery_assetOrError_AssetNotFoundError {
  __typename: "AssetNotFoundError";
}

export interface AssetEntryRootQuery_assetOrError_AssetsNotSupportedError {
  __typename: "AssetsNotSupportedError";
  message: string;
}

export interface AssetEntryRootQuery_assetOrError_Asset_key {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetEntryRootQuery_assetOrError_Asset {
  __typename: "Asset";
  key: AssetEntryRootQuery_assetOrError_Asset_key;
}

export type AssetEntryRootQuery_assetOrError = AssetEntryRootQuery_assetOrError_AssetNotFoundError | AssetEntryRootQuery_assetOrError_AssetsNotSupportedError | AssetEntryRootQuery_assetOrError_Asset;

export interface AssetEntryRootQuery {
  assetOrError: AssetEntryRootQuery_assetOrError;
}

export interface AssetEntryRootQueryVariables {
  assetKey: AssetKeyInput;
}
