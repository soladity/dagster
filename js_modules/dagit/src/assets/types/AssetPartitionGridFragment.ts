// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: AssetPartitionGridFragment
// ====================================================

export interface AssetPartitionGridFragment_runOrError_PipelineRunNotFoundError {
  __typename: "PipelineRunNotFoundError" | "PythonError";
}

export interface AssetPartitionGridFragment_runOrError_PipelineRun {
  __typename: "PipelineRun";
  pipelineSnapshotId: string | null;
}

export type AssetPartitionGridFragment_runOrError = AssetPartitionGridFragment_runOrError_PipelineRunNotFoundError | AssetPartitionGridFragment_runOrError_PipelineRun;

export interface AssetPartitionGridFragment {
  __typename: "AssetMaterialization";
  partition: string | null;
  runOrError: AssetPartitionGridFragment_runOrError;
}
