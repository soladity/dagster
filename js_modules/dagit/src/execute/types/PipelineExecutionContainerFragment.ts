// @generated
/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: PipelineExecutionContainerFragment
// ====================================================

export interface PipelineExecutionContainerFragment_PipelineNotFoundError {
  __typename: "PipelineNotFoundError" | "PythonError";
}

export interface PipelineExecutionContainerFragment_Pipeline_modes {
  __typename: "Mode";
  name: string;
  description: string | null;
}

export interface PipelineExecutionContainerFragment_Pipeline {
  __typename: "Pipeline";
  name: string;
  modes: PipelineExecutionContainerFragment_Pipeline_modes[];
}

export interface PipelineExecutionContainerFragment_InvalidSubsetError_pipeline_modes {
  __typename: "Mode";
  name: string;
  description: string | null;
}

export interface PipelineExecutionContainerFragment_InvalidSubsetError_pipeline {
  __typename: "Pipeline";
  name: string;
  modes: PipelineExecutionContainerFragment_InvalidSubsetError_pipeline_modes[];
}

export interface PipelineExecutionContainerFragment_InvalidSubsetError {
  __typename: "InvalidSubsetError";
  message: string;
  pipeline: PipelineExecutionContainerFragment_InvalidSubsetError_pipeline;
}

export type PipelineExecutionContainerFragment = PipelineExecutionContainerFragment_PipelineNotFoundError | PipelineExecutionContainerFragment_Pipeline | PipelineExecutionContainerFragment_InvalidSubsetError;
