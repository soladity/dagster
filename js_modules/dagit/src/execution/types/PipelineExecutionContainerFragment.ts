

/* tslint:disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: PipelineExecutionContainerFragment
// ====================================================

export interface PipelineExecutionContainerFragment_runs_logs_nodes_LogMessageEvent {
  message: string;
  timestamp: string;
  level: LogLevel;
  __typename: "LogMessageEvent" | "PipelineStartEvent" | "PipelineSuccessEvent" | "PipelineFailureEvent" | "PipelineProcessStartEvent";
}

export interface PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepFailureEvent_step {
  name: string;
}

export interface PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepFailureEvent_error {
  stack: string[];
  message: string;
}

export interface PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepFailureEvent {
  message: string;
  timestamp: string;
  level: LogLevel;
  __typename: "ExecutionStepFailureEvent";
  step: PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepFailureEvent_step;
  error: PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepFailureEvent_error;
}

export interface PipelineExecutionContainerFragment_runs_logs_nodes_PipelineProcessStartedEvent {
  message: string;
  timestamp: string;
  level: LogLevel;
  __typename: "PipelineProcessStartedEvent";
  processId: number;
}

export interface PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepStartEvent_step {
  name: string;
}

export interface PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepStartEvent {
  message: string;
  timestamp: string;
  level: LogLevel;
  __typename: "ExecutionStepStartEvent" | "ExecutionStepSuccessEvent";
  step: PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepStartEvent_step;
}

export type PipelineExecutionContainerFragment_runs_logs_nodes = PipelineExecutionContainerFragment_runs_logs_nodes_LogMessageEvent | PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepFailureEvent | PipelineExecutionContainerFragment_runs_logs_nodes_PipelineProcessStartedEvent | PipelineExecutionContainerFragment_runs_logs_nodes_ExecutionStepStartEvent;

export interface PipelineExecutionContainerFragment_runs_logs_pageInfo {
  lastCursor: any | null;
}

export interface PipelineExecutionContainerFragment_runs_logs {
  nodes: PipelineExecutionContainerFragment_runs_logs_nodes[];
  pageInfo: PipelineExecutionContainerFragment_runs_logs_pageInfo;
}

export interface PipelineExecutionContainerFragment_runs_executionPlan_steps_solid {
  name: string;
}

export interface PipelineExecutionContainerFragment_runs_executionPlan_steps {
  name: string;
  solid: PipelineExecutionContainerFragment_runs_executionPlan_steps_solid;
  kind: StepKind;
}

export interface PipelineExecutionContainerFragment_runs_executionPlan {
  steps: PipelineExecutionContainerFragment_runs_executionPlan_steps[];
}

export interface PipelineExecutionContainerFragment_runs {
  runId: string;
  status: PipelineRunStatus;
  logs: PipelineExecutionContainerFragment_runs_logs;
  executionPlan: PipelineExecutionContainerFragment_runs_executionPlan;
}

export interface PipelineExecutionContainerFragment_environmentType {
  key: string;
}

export interface PipelineExecutionContainerFragment_configTypes_EnumConfigType {
  __typename: "EnumConfigType" | "RegularConfigType" | "ListConfigType" | "NullableConfigType";
  key: string;
  name: string | null;
  isSelector: boolean;
}

export interface PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields_configType_EnumConfigType {
  __typename: "EnumConfigType" | "CompositeConfigType" | "RegularConfigType" | "NullableConfigType";
  key: string;
  name: string | null;
  isList: boolean;
  isNullable: boolean;
}

export interface PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields_configType_ListConfigType_ofType {
  key: string;
}

export interface PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields_configType_ListConfigType {
  __typename: "ListConfigType";
  key: string;
  name: string | null;
  isList: boolean;
  isNullable: boolean;
  ofType: PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields_configType_ListConfigType_ofType;
}

export type PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields_configType = PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields_configType_EnumConfigType | PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields_configType_ListConfigType;

export interface PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields {
  name: string;
  isOptional: boolean;
  configType: PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields_configType;
}

export interface PipelineExecutionContainerFragment_configTypes_CompositeConfigType {
  __typename: "CompositeConfigType";
  key: string;
  name: string | null;
  isSelector: boolean;
  fields: PipelineExecutionContainerFragment_configTypes_CompositeConfigType_fields[];
}

export type PipelineExecutionContainerFragment_configTypes = PipelineExecutionContainerFragment_configTypes_EnumConfigType | PipelineExecutionContainerFragment_configTypes_CompositeConfigType;

export interface PipelineExecutionContainerFragment {
  name: string;
  runs: PipelineExecutionContainerFragment_runs[];
  environmentType: PipelineExecutionContainerFragment_environmentType;
  configTypes: PipelineExecutionContainerFragment_configTypes[];
}

/* tslint:disable */
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum EvaluationErrorReason {
  FIELD_NOT_DEFINED = "FIELD_NOT_DEFINED",
  MISSING_REQUIRED_FIELD = "MISSING_REQUIRED_FIELD",
  RUNTIME_TYPE_MISMATCH = "RUNTIME_TYPE_MISMATCH",
  SELECTOR_FIELD_ERROR = "SELECTOR_FIELD_ERROR",
}

export enum LogLevel {
  CRITICAL = "CRITICAL",
  DEBUG = "DEBUG",
  ERROR = "ERROR",
  INFO = "INFO",
  WARNING = "WARNING",
}

/**
 * An enumeration.
 */
export enum PipelineRunStatus {
  FAILURE = "FAILURE",
  NOT_STARTED = "NOT_STARTED",
  STARTED = "STARTED",
  SUCCESS = "SUCCESS",
}

export enum StepKind {
  INPUT_EXPECTATION = "INPUT_EXPECTATION",
  INPUT_THUNK = "INPUT_THUNK",
  JOIN = "JOIN",
  MATERIALIZATION_THUNK = "MATERIALIZATION_THUNK",
  OUTPUT_EXPECTATION = "OUTPUT_EXPECTATION",
  SERIALIZE = "SERIALIZE",
  TRANSFORM = "TRANSFORM",
}

/**
 * This type represents the fields necessary to identify a
 *         pipeline or pipeline subset.
 */
export interface ExecutionSelector {
  name: string;
  solidSubset?: string[] | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================