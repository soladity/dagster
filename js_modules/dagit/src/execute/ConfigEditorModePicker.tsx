import * as React from "react";
import { Button, Intent, Menu } from "@blueprintjs/core";
import { IconNames } from "@blueprintjs/icons";
import { Select } from "@blueprintjs/select";
import { PipelineDetailsFragment_modes } from "./types/PipelineDetailsFragment";
import { ModeNotFoundError } from "./ExecutionSessionContainer";

interface ConfigEditorModePickerProps {
  modes: PipelineDetailsFragment_modes[];
  modeError?: ModeNotFoundError;
  modeName: string | null;
  onModeChange: (mode: string) => void;
}

interface Mode {
  name: string;
}

const ModeSelect = Select.ofType<Mode>();

export class ConfigEditorModePicker extends React.PureComponent<
  ConfigEditorModePickerProps
> {
  getModeFromProps = (props: ConfigEditorModePickerProps) => {
    return props.modeName
      ? props.modes.find(m => m.name === props.modeName)
      : props.modes[0];
  };

  getCurrentMode = () => {
    return this.getModeFromProps(this.props);
  };

  componentDidMount() {
    const currentMode = this.getCurrentMode();
    if (currentMode) {
      this.props.onModeChange && this.props.onModeChange(currentMode.name);
    }
  }

  componentDidUpdate(prevProps: ConfigEditorModePickerProps) {
    const currentMode = this.getCurrentMode();
    const prevMode = this.getModeFromProps(prevProps);

    if (currentMode && currentMode !== prevMode) {
      this.props.onModeChange && this.props.onModeChange(currentMode.name);
    }
  }

  render() {
    const singleMode = this.props.modes.length === 1;
    const currentMode = this.getCurrentMode();
    const valid = !this.props.modeError;

    return (
      <ModeSelect
        activeItem={currentMode}
        filterable={true}
        disabled={singleMode && valid}
        items={this.props.modes}
        itemPredicate={(query, mode) =>
          query.length === 0 || mode.name.includes(query)
        }
        itemRenderer={(mode, props) => (
          <Menu.Item
            active={props.modifiers.active}
            key={mode.name}
            text={mode.name}
            onClick={props.handleClick}
          />
        )}
        onItemSelect={this.onItemSelect}
      >
        <Button
          icon={valid ? "insert" : IconNames.WARNING_SIGN}
          intent={valid ? Intent.NONE : Intent.WARNING}
          text={
            valid
              ? currentMode
                ? `Mode: ${currentMode.name}`
                : "Select Mode"
              : "Invalid Mode Selection"
          }
          disabled={singleMode && valid}
          rightIcon="caret-down"
        />
      </ModeSelect>
    );
  }

  private onItemSelect = (mode: Mode) => {
    this.props.onModeChange && this.props.onModeChange(mode.name);
  };
}
