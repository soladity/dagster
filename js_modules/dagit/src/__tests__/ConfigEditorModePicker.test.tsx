import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import ConfigEditorModePicker from "../execute/ConfigEditorModePicker";
import { PipelineExecutionContainerFragment } from "../execute/types/PipelineExecutionContainerFragment";

it("renders single mode pipelines", () => {
  const singleModePipelineData: PipelineExecutionContainerFragment = {
    __typename: "Pipeline",
    configTypes: [],
    environmentType: {
      __typename: "CompositeConfigType",
      key: "SingleModePipeline.Mode.Default.Environment"
    },
    modes: [
      {
        __typename: "Mode",
        description: null,
        name: "default"
      }
    ],
    name: "single_mode_pipeline"
  };
  const componentNullSelected = TestRenderer.create(
    <BrowserRouter>
      <ConfigEditorModePicker
        pipeline={singleModePipelineData}
        modeName={null}
        onModeChange={() => null}
      />
    </BrowserRouter>
  );
  expect(componentNullSelected.toJSON()).toMatchSnapshot();

  const componentDefaultSelected = TestRenderer.create(
    <BrowserRouter>
      <ConfigEditorModePicker
        pipeline={singleModePipelineData}
        modeName={"default"}
        onModeChange={() => null}
      />
    </BrowserRouter>
  );
  expect(componentDefaultSelected.toJSON()).toMatchSnapshot();
});

it("renders multi mode pipelines", () => {
  const multiModePipelineData: PipelineExecutionContainerFragment = {
    __typename: "Pipeline",
    configTypes: [],
    environmentType: {
      __typename: "CompositeConfigType",
      key: "MultiModePipeline.Mode.Default.Environment"
    },
    modes: [
      {
        __typename: "Mode",
        description: "Mode 1",
        name: "mode_1"
      },
      {
        __typename: "Mode",
        description: "Mode 2",
        name: "mode_2"
      }
    ],
    name: "multi_mode_pipeline"
  };
  const componentNullSelected = TestRenderer.create(
    <BrowserRouter>
      <ConfigEditorModePicker
        pipeline={multiModePipelineData}
        modeName={null}
        onModeChange={() => null}
      />
    </BrowserRouter>
  );
  expect(componentNullSelected.toJSON()).toMatchSnapshot();

  const componentMode1Selected = TestRenderer.create(
    <BrowserRouter>
      <ConfigEditorModePicker
        pipeline={multiModePipelineData}
        modeName={"mode1"}
        onModeChange={() => null}
      />
    </BrowserRouter>
  );
  expect(componentMode1Selected.toJSON()).toMatchSnapshot();
});
