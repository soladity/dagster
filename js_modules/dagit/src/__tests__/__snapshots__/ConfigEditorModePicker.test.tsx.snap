// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error mode 1`] = `
<span
  className="bp3-popover-wrapper"
>
  <span
    className="bp3-popover-target"
    onClick={[Function]}
  >
    <div
      className=""
      onKeyDown={[Function]}
    >
      <button
        className="bp3-button bp3-intent-warning"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-warning-sign"
          icon="warning-sign"
        >
          <svg
            data-icon="warning-sign"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              warning-sign
            </desc>
            <path
              d="M15.84 13.5l.01-.01-7-12-.01.01c-.17-.3-.48-.5-.85-.5s-.67.2-.85.5l-.01-.01-7 12 .01.01c-.09.15-.15.31-.15.5 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 0-.19-.06-.35-.15-.5zm-6.85-.51h-2v-2h2v2zm0-3h-2v-5h2v5z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Invalid Mode Selection
        </span>
        <span
          className="bp3-icon bp3-icon-caret-down"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </span>
</span>
`;

exports[`renders multi mode pipelines 1`] = `
<span
  className="bp3-popover-wrapper"
>
  <span
    className="bp3-popover-target"
    onClick={[Function]}
  >
    <div
      className=""
      onKeyDown={[Function]}
    >
      <button
        className="bp3-button"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-insert"
          icon="insert"
        >
          <svg
            data-icon="insert"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              insert
            </desc>
            <path
              d="M5 9h2v2c0 .6.4 1 1 1s1-.4 1-1V9h2c.6 0 1-.4 1-1s-.4-1-1-1H9V5c0-.6-.4-1-1-1s-1 .4-1 1v2H5c-.6 0-1 .4-1 1s.4 1 1 1zm10-9H1C.4 0 0 .4 0 1v14c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V1c0-.6-.4-1-1-1zm-1 14H2V2h12v12z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Mode: mode_1
        </span>
        <span
          className="bp3-icon bp3-icon-caret-down"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </span>
</span>
`;

exports[`renders multi mode pipelines 2`] = `
<span
  className="bp3-popover-wrapper"
>
  <span
    className="bp3-popover-target"
    onClick={[Function]}
  >
    <div
      className=""
      onKeyDown={[Function]}
    >
      <button
        className="bp3-button"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-insert"
          icon="insert"
        >
          <svg
            data-icon="insert"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              insert
            </desc>
            <path
              d="M5 9h2v2c0 .6.4 1 1 1s1-.4 1-1V9h2c.6 0 1-.4 1-1s-.4-1-1-1H9V5c0-.6-.4-1-1-1s-1 .4-1 1v2H5c-.6 0-1 .4-1 1s.4 1 1 1zm10-9H1C.4 0 0 .4 0 1v14c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V1c0-.6-.4-1-1-1zm-1 14H2V2h12v12z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Mode: mode_1
        </span>
        <span
          className="bp3-icon bp3-icon-caret-down"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </span>
</span>
`;

exports[`renders single mode pipelines 1`] = `
<span
  className="bp3-popover-wrapper"
>
  <span
    className="bp3-popover-target"
    onClick={[Function]}
  >
    <div
      className=""
      onKeyDown={[Function]}
    >
      <button
        className="bp3-button bp3-disabled"
        disabled={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-insert"
          icon="insert"
        >
          <svg
            data-icon="insert"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              insert
            </desc>
            <path
              d="M5 9h2v2c0 .6.4 1 1 1s1-.4 1-1V9h2c.6 0 1-.4 1-1s-.4-1-1-1H9V5c0-.6-.4-1-1-1s-1 .4-1 1v2H5c-.6 0-1 .4-1 1s.4 1 1 1zm10-9H1C.4 0 0 .4 0 1v14c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V1c0-.6-.4-1-1-1zm-1 14H2V2h12v12z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Mode: default
        </span>
        <span
          className="bp3-icon bp3-icon-caret-down"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </span>
</span>
`;

exports[`renders single mode pipelines 2`] = `
<span
  className="bp3-popover-wrapper"
>
  <span
    className="bp3-popover-target"
    onClick={[Function]}
  >
    <div
      className=""
      onKeyDown={[Function]}
    >
      <button
        className="bp3-button bp3-disabled"
        disabled={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-insert"
          icon="insert"
        >
          <svg
            data-icon="insert"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              insert
            </desc>
            <path
              d="M5 9h2v2c0 .6.4 1 1 1s1-.4 1-1V9h2c.6 0 1-.4 1-1s-.4-1-1-1H9V5c0-.6-.4-1-1-1s-1 .4-1 1v2H5c-.6 0-1 .4-1 1s.4 1 1 1zm10-9H1C.4 0 0 .4 0 1v14c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V1c0-.6-.4-1-1-1zm-1 14H2V2h12v12z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Mode: default
        </span>
        <span
          className="bp3-icon bp3-icon-caret-down"
          icon="caret-down"
        >
          <svg
            data-icon="caret-down"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              caret-down
            </desc>
            <path
              d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </span>
</span>
`;
